spec/registration_system_spec.rb

```diff
@@ -1,15 +1,32 @@
-describe LinuxAdmin::RegistrationSystem do
+shared_context "RegistrationSystem.registration_type stubbing", :registered_system_stubbing do
+  # stub out `warn` messages from LinuxAdmin::Rhn
+  before do
+    allow_any_instance_of(LinuxAdmin::Rhn).to receive(:warn)
+  end
+
+  def stub_registered_to_system(*system)
+    allow_any_instance_of(LinuxAdmin::SubscriptionManager).to receive_messages(:registered? => system.include?(:sm))
```

Comment:
Can we use something other than `allow_any_instance_of` here?

I think `allow(LinuxAdmin::SubscriptionManager).to receive(:new).and_return(double("SubscriptionManager", :registered? => system.include?(:sm))` would do the job as well, right?

Reply:
Ah, just saw this was only moved, thought this was new.

Reply:
Yeah... I wasn't a fan of it myself, but I figured changing it now was not a good idea.